open-craft/xblock-group-project-v2
# Dashboard details view

**Configuration**
Add the following XBlock settings to lms.env.json
```
"XBLOCK_SETTINGS": {
        "group_project_v2": {
            "dashboard_details_url": "/admin/workgroup/dashboard/programs/{program_id}/courses/{course_id}/projects/{project_id}/details?activate_block_id={activity_id}",
            "ta_review_url": "/courses/{course_id}/group_work/{group_id}?activate_block_id={activate_block_id}"
        }
    },
```

**Testing instructions:**
1. Configure GP so that at least one group have more than one participant
2. Complete non-TA graded Peer Grading stages by *some* users
3. Log in as McKA TA (username mcka_ta_user) or internal/company/McKA admin
4. Go to Admin dashboard -> Group Work -> Access V2 Dashboard
5. Choose program, course and project
6. Click on blue > in any stage that has it in bottom-right corner
7. You should arrive to Details view.
8. Should display all graded stages in an activity (only Peer Grading stages are considered graded)
9. Clicking email icon in group header should download a csv report containing users who haven't completed stage yet (should match what's displayed in this view)
10. Groups are collapsible
11. User emails are mailto: links
12. Peer Grading stages (i.e. all stages visible) should display which groups user is to grade.
13. Group labels are links to TA grading view. Groups in left column take to first Peer Grading stage in activity, groups in stage columns take exactly to that stage.
14. Most exciting part to test: data displayed should match actual completion stats. Completion criteria for groups: Completed - all users have completed, not started - all users have not started the stage, otherwise partially complete.
15. Clicking "Back to Group Work V2 Dashboard" should get you back to previous page, and restore filters (in a very fancy way - you may impress others by pretending you have a telepathic interface)

Big red failure condition: if any group/user displays red stop sign instead of usual completion data - something went wrong and stage state data for that user have not been calculated.
